app/src/org/commcare/dalvik/activities/RecoveryActivity.java

```diff
@@ -53,10 +53,8 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         
-        if(this.getDestroyedActivityState() != null) {
-            //We just rotated or whatever, don't re-initialize everything
-        } else {
-            //Fresh Start, statewise.
+        if (savedInstanceState == null) {

```

Comment:
Just as a heads up, savedInstanceState is not a reliable mechanism and its use is discouraged, so I wouldn't generally use it as a flag for whether this is restarting.